src/v/cluster/config_frontend.cc

```diff
@@ -42,7 +42,12 @@ ss::future<config_frontend::patch_result> config_frontend::patch(
           .errc = errc::no_leader_controller, .version = config_version_unset};
     }
     if (leader == _self) {
-        co_return co_await do_patch(std::move(update), timeout);
+        co_return co_await container().invoke_on(
```

Comment:
why are we switching cores here to invoke `do_patch` rather than requiring the caller to invoke `patch` on the correct core, like in `cluster::service::config_update`?

to be clear, i'm not saying it should be one way or another. but i do think it should be consistent. i'd probably choose whichever pattern aligned with a majority of the callers, and then add a comment expressing the expectation on callers (if there are any), and an assertion to check it.

Reply:
It's clearly been used incorrectly; I can't imagine a downside of making `do_,patch` private and dispatching to the correct core within `patch`, but I may have missed something.

The primary advantage is that it makes the AP  harder to misuse.

I may be missing something, though.

Reply:
Agree with what Ben said above.

This also improves consistency locally, by making all of the public methods of `config_frontend` callable from any shard, not just `set_status`.

Globally across `*_frontend.h` public methods, there is an inconsistency where some methods are callable from any shard while some aren't. I would think that since `invoke_on` seems cheap, we should standardise making it the `*_frontend` classes' responsibility to delegate work to the correct core. But that's a larger undertaking.

Reply:
Yes, allowing the method to be invoked on any core is great.

Internally we should also be consistent:

```
    auto leader = _leaders.local().get_leader(model::controller_ntp);
    if (!leader) {
        co_return patch_result{
          .errc = errc::no_leader_controller, .version = config_version_unset};
    }
    if (leader == _self) {
        co_return co_await do_patch(std::move(update), timeout);
        co_return co_await container().invoke_on(
```

here we are combining state from different cores. it's benign in this case, but in general, it should be consistent.